### PR TITLE
add cloudflare feed

### PR DESCRIPTION
**Pull request by ronykoz**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

Added Cloudflare feed

### CODE REVIEW

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  name: url
+  options:
+  - https://www.cloudflare.com/ips-v4
+  - https://www.cloudflare.com/ips-v6
+  required: true
+  type: 16
+- display: Fetch indicators
+  name: feed
+  required: false
+  type: 8
```
Comment (anara123) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]: add `defaultValue: true`

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  - A - Completely reliable
+  - B - Usually reliable
+  - C - Fairly reliable
+  - D - Not usually reliable
+  - E - Unreliable
+  - F - Reliability cannot be judged
+  required: false
+  type: 15
+- defaultvalue: indicatorType
+  hidden: true
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]: Should remove `hidden` fields.

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+- additionalinfo: Timeout of the polling request in seconds.
+  defaultvalue: '20'
+  display: Request Timeout
+  name: polling_timeout
+  required: false
+  type: 0
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]: Move `polling_timeout` parameter above `proxy` and `insecure`, please.

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  - D - Not usually reliable
+  - E - Unreliable
+  - F - Reliability cannot be judged
+  required: false
+  type: 15
+- defaultvalue: indicatorType
+  hidden: true
+  name: feedExpirationPolicy
+  required: false
+  type: 0
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
- display: ""
  name: feedExpirationPolicy
  defaultvalue: indicatorType
  type: 17
  required: false
  options:
  - never
  - interval
  - indicatorType
  - suddenDeath
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+- hidden: true
+  name: feedExpirationInterval
+  required: false
+  type: 1
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
- display: ""
  name: feedExpirationInterval
  defaultvalue: "20160"
  type: 1
  required: false
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  required: false
+  type: 8
+- additionalinfo: Timeout of the polling request in seconds.
+  defaultvalue: '20'
+  display: Request Timeout
+  name: polling_timeout
+  required: false
+  type: 0
+description: Use the Cloudflare feed integration to fetch indicators from the feed.
+display: Cloudflare feed
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
display: Cloudflare Feed
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+name: Cloudflare feed
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
name: Cloudflare Feed
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
+category: Data Enrichment & Threat Intelligence
+commonfields:
+  id: Cloudflare feed
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
  id: Cloudflare Feed
```

File: Tests/conf.json [hunk 9]
```diff
@@ -7,6 +7,11 @@
             "playbookID": "Blocklist_de - Test",
             "fromversion": "5.5.0"
         },
+        {
+            "integrations": "Cloudflare feed",
```
Comment (avidan-H) [inline Tests/conf.json]:
```suggestion
            "integrations": "Cloudflare Feed",
```

**Comment by ronykoz** [comment]
Added docs-approved as there is no docs to review here

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+  - https://www.cloudflare.com/ips-v6
+  required: true
+  type: 16
+- display: Fetch indicators
+  name: feed
+  required: false
+  type: 8
+  defaultvalue: true
+- defaultvalue: '5'
+  display: Fetch Interval
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
  display: Feed Fetch Interval
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+  name: feedFetchInterval
+  required: false
+  type: 19
+- additionalinfo: Indicators from this integration instance will be marked with this
+    reputation.
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
    reputation
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+  defaultvalue: Good
+  display: Indicator reputation
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
  display: Indicator Reputation
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+  name: feedReputation
+  options:
+  - None
+  - Good
+  - Suspicious
+  - Bad
+  required: false
+  type: 18
+- defaultvalue: A - Completely reliable
+  display: Reliability
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
  display: Source Reliability
```

File: Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+  name: feedReliability
+  options:
+  - A - Completely reliable
+  - B - Usually reliable
+  - C - Fairly reliable
+  - D - Not usually reliable
+  - E - Unreliable
+  - F - Reliability cannot be judged
+  required: false
+  type: 15
```
Comment (avidan-H) [inline Packs/FeedCloudflare/Integrations/FeedCloudflare/FeedCloudflare.yml]:
```suggestion
  type: 15
  additionalinfo: Reliability of the source providing the intelligence data
```

File: Packs/FeedCloudflare/TestPlaybooks/playbook-FeedCloudflare-test.yml [hunk 15]
```diff
@@ -0,0 +1,186 @@
+elasticcommonfields: {}
+id: cloudflare - Test
+version: 5
```
Comment (avidan-H) [inline Packs/FeedCloudflare/TestPlaybooks/playbook-FeedCloudflare-test.yml]:
```suggestion
version: -1
```

File: Packs/FeedCloudflare/TestPlaybooks/playbook-FeedCloudflare-test.yml [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+        "height": 735,
+        "width": 380,
+        "x": 50,
+        "y": 50
+      }
+    }
+  }
+inputs: []
+outputs: []
+sourceplaybookid: cloudflare - Test
```
Comment (avidan-H) [inline Packs/FeedCloudflare/TestPlaybooks/playbook-FeedCloudflare-test.yml]:
```suggestion
sourceplaybookid: cloudflare - Test
fromversion: 5.5.0
```

**Comment by ronykoz** [comment]
pushing due to unrelated failures

**Pull request merged**